packages/polling-controller/src/StaticIntervalPollingController.test.ts

```diff
@@ -234,6 +273,19 @@ describe('StaticIntervalPollingController', () => {
         ['sepolia', { address: '0x2' }],
       ]);
     });
+
+    it('should stop polling session after current iteration if stop is requested while current iteration is still executing', async () => {
+      const pollingToken = controller.startPollingByNetworkClientId('mainnet');
+      await advanceTime({ clock, duration: 0 });
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      controller.stopPollingByPollingToken(pollingToken);
+      controller.executePollPromises[0].resolve();
+      await advanceTime({ clock, duration: TICK_TIME });
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      await advanceTime({ clock, duration: TICK_TIME });
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      controller.stopAllPolling();
+    });
```

Comment:
Not blocking but should we do a version of this test where we have multiple polling keys running?

Reply:
not sure if necessary since we have other tests that specifically test that stopping polling for one key does not stop polling for other keys

Reply:
Fair yeah I guess I'm wondering about a test like starting polling sessions for a few keys and then calling stopAllPolling before execution resolves for both.